src/zh/manhuaren/src/eu/kanade/tachiyomi/extension/zh/manhuaren/Manhuaren.kt

```diff
@@ -35,28 +43,37 @@ import javax.crypto.Cipher
 import kotlin.random.Random
 import kotlin.random.nextUBytes
 
-class Manhuaren : HttpSource() {
+class Manhuaren : HttpSource(), ConfigurableSource {
     override val lang = "zh"
     override val supportsLatest = true
     override val name = "漫画人"
     override val baseUrl = "http://mangaapi.manhuaren.com"
 
     private val pageSize = 20
     private val baseHttpUrl = baseUrl.toHttpUrl()
+    private val preferences: SharedPreferences =
+        Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
 
     private val gsnSalt = "4e0a48e1c0b54041bce9c8f0e036124d"
     private val encodedPublicKey = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAmFCg289dTws27v8GtqIffkP4zgFR+MYIuUIeVO5AGiBV0rfpRh5gg7i8RrT12E9j6XwKoe3xJz1khDnPc65P5f7CJcNJ9A8bj7Al5K4jYGxz+4Q+n0YzSllXPit/Vz/iW5jFdlP6CTIgUVwvIoGEL2sS4cqqqSpCDKHSeiXh9CtMsktc6YyrSN+8mQbBvoSSew18r/vC07iQiaYkClcs7jIPq9tuilL//2uR9kWn5jsp8zHKVjmXuLtHDhM9lObZGCVJwdlN2KDKTh276u/pzQ1s5u8z/ARtK26N8e5w8mNlGcHcHfwyhjfEQurvrnkqYH37+12U3jGk5YNHGyOPcwIDAQAB"
     private val imei: String by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) { generateIMEI() }
     private val token: String by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) { fetchToken() }
-    private var userId = "-1"
-    private var lastUsedTime = ""
+    private var userId: String = preferences.getString(USER_ID_PREF, null) ?: "-1"
+    private val lastUsedTime: String by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) { generateLastUsedTime() }
```

Comment:
```suggestion
    private val lastUsedTime: String by lazy { generateLastUsedTime() }
```

The default mode *is* synchronized.